### ISSUE TITLE
Feature Request: Write-Progress display in the Windows Notification Center

### ISSUE THREAD

**Issue by Jack-Works** [PowerShell/PowerShell#13072]
## Summary of the new feature/enhancement

Write-Progress can display its progress like this

<img src="https://docs.microsoft.com/en-us/windows/uwp/design/shell/tiles-and-notifications/images/toast-progressbar-annotated.png" />

This can be opt-in so it's not breaking the existing program.

**Comment by vexx32** [comment]
It's not a builtin, but I believe the BurntToast module would be capable of something like this.

**Comment by rjmholt** [comment]
This sounds like something that's probably better implemented as a module

**Comment by Jack-Works** [comment]
Does it possible to rewrite the built-in Write-Progress from the user land?

**Comment by rjmholt** [comment]
Various possibilities are:

- Write a new PSHost (a lot of work) that hooks the underlying progress API up to this. Probably not ideal here
- Override `Write-Progress` with your own command that redirects progress to the WNC (this would probably break a number of applications depending on `Write-Progress` to write to the host)
- Write a new module and define something like `Write-WncProgress` to allow both to be used side-by-side

**Comment by iSazonov** [comment]
Blocked by #13042

**Comment by rjmholt** [comment]
I’m going to close this feature request as out-of-scope for the PowerShell project, noting @iSazonov’s point about required dependencies
